### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by harrison-blake**

### CODE REVIEW

File: test/sales_engine_test.rb [hunk 1]
```diff
@@ -3,25 +3,48 @@
...
+    assert_instance_of Array, @se.pass_item_array
+    assert_equal 1367, @se.count_items
+  end
+  def test_count_merchants
+    assert_instance_of Array, @se.pass_item_array
+    assert_equal 475, @se.count_merchants
+  end
+  def test_find_all_by_merchant_id
+    assert_instance_of Array, @se.find_all_by_merchant_id(12_334_141)
+    assert_equal 263395237, @se.find_all_by_merchant_id(12_334_141)[0].id
```
[the remark this answers is not in the document]
Comment (domo2192) [reply test/sales_engine_test.rb:48]: Rubocop will fix this

File: lib/mathable.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module Mathable
+  def average_stuff(arg_1, arg_2)
```
Comment (arikalea) [inline lib/mathable.rb:2]: Looks good, we'll need to fix the spacing between methods.

File: lib/invoice.rb [hunk 3]
```diff
@@ -1,16 +1,17 @@
 class Invoice
 attr_reader :id,
             :created_at,
-            :updated_at,
             :customer_id,
-            :merchant_id,
-            :status
+            :merchant_id
+            
+attr_accessor :status,
```
Comment (domo2192) [inline lib/invoice.rb:7]: Do you want these to be attr accessor?

File: lib/sales_analyst.rb [hunk 4]
```diff
@@ -1,94 +1,72 @@
...
+  def items_per_merchant
+    sales_engine.items
+  end
+  def average_item_price
+    sales_engine.average_item_price
+  end
+  def item_price_standard_deviation
+    average = average_item_price
+    total = sales_engine.pass_item_array.sum do |item|
+      (average - item.unit_price_to_dollars)**2
```
Comment (Trevor-Robinson) [inline lib/sales_analyst.rb:31]: Looks good. Might be able to break out into math module

Comment (domo2192) [reply lib/sales_analyst.rb:31]: Good idea

File: lib/mathable.rb (hunk 2 above)
Comment (domo2192) [reply lib/mathable.rb:2]: agreed

File: lib/sales_analyst.rb [hunk 5]
```diff
@@ -1,94 +1,72 @@
...
-    price_total = sales_engine.items.all.sum do |item|
-      item.unit_price_to_dollars
-    end
-    (price_total/ total_items).round(2)
-  end
-
-  def item_price_standard_deviation
-    average = average_item_price
-    total = sales_engine.items.all.sum do |item|
-      (average - item.unit_price_to_dollars)**2
```
Comment (Trevor-Robinson) [inline lib/sales_analyst.rb:65]: Nice job! Might be able to move to mathable module.

File: lib/item_repository.rb [hunk 6]
```diff
@@ -1,91 +1,108 @@
 require_relative 'item'
 require_relative 'sales_engine'
+require_relative 'mathable'
```
Comment (harrison-blake) [inline lib/item_repository.rb:3]: add space between requires and  class def

**Pull request merged**